assets/js/modules/analytics-4/components/common/EnhancedMeasurementSwitch.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * EnhancedMeasurementSwitch component.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { createInterpolateElement, useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { ProgressBar, Switch } from 'googlesitekit-components';
+import { CORE_FORMS } from '../../../../googlesitekit/datastore/forms/constants';
+import {
+	ENHANCED_MEASUREMENT_ENABLED,
+	ENHANCED_MEASUREMENT_FORM,
+} from '../../datastore/constants';
+import SupportLink from '../../../../components/SupportLink';
+const { useSelect, useDispatch } = Data;
+
+export default function EnhancedMeasurementSwitch( {
+	onClick,
+	disabled = false,
+	loading = false,
+} ) {
+	const isEnhancedMeasurementEnabled = useSelect( ( select ) =>
+		select( CORE_FORMS ).getValue(
+			ENHANCED_MEASUREMENT_FORM,
+			ENHANCED_MEASUREMENT_ENABLED
+		)
+	);
+
+	const { setValues } = useDispatch( CORE_FORMS );
+
+	const handleClick = useCallback( () => {
+		setValues( ENHANCED_MEASUREMENT_FORM, {
+			[ ENHANCED_MEASUREMENT_ENABLED ]: ! isEnhancedMeasurementEnabled,
+		} );
+		onClick?.();
```

Comment:
Although not in the IB, it was necessary to add a click handler here in order to be able to update the data stream `streamEnabled` setting in local state in `SettingsEnhancedMeasurementSwitch`.